activerecord/lib/active_record/type/decimal.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Type
+    class Decimal < ActiveModel::Type::Decimal # :nodoc:
+      def serialize(value)
+        raise ActiveModel::RangeError, "cannot be infinite" if value&.to_d&.infinite?
```

Comment:
Calling `to_d` here is unnecessary: the value will have already been converted to a BigDecimal by `cast_value`.

Reply:
This appears to sometimes be a string or nil according to the test suite (`DefaultNumbersTest#test_default_decimal_number`, `DefaultNumbersTest#test_default_positive_integer`, `SchemaDumperDefaultsTest#test_schema_dump_with_float_column_infinity_default`), so I used `#to_d` here. It appears to be because `#serialize` is used for quoting here: `active_record/connection_adapters/abstract/quoting.rb:92`